packages/utils/src/eslint-utils/rule-tester/RuleTester.ts

```diff
@@ -107,9 +125,48 @@ class RuleTester extends TSESLint.RuleTester {
   // This is a lot more explicit
   run<TMessageIds extends string, TOptions extends Readonly<unknown[]>>(
     name: string,
-    rule: TSESLint.RuleModule<TMessageIds, TOptions>,
+    rule: RuleModule<TMessageIds, TOptions>,
     testsReadonly: RunTests<TMessageIds, TOptions>,
   ): void {
+    if (
+      this.#baseOptions.dependencyConstraints &&
+      !satisfiesAllDependencyConstraints(
+        this.#baseOptions.dependencyConstraints,
+      )
+    ) {
+      type DescribeWithMaybeSkip = RuleTesterTestFrameworkFunction & {
+        skip?: RuleTesterTestFrameworkFunction;
+      };
+      const describe: DescribeWithMaybeSkip = this.staticThis.describe;
+      if ('skip' in describe && typeof describe.skip === 'function') {
```

Comment:
Non-blocking suggestion: maybe extract this into a type predicate to make it more clear? Then I suspect you can remove the `as DescribeWithSkip` later on.